File: app/src/main/java/com/gustu/logbook/main/adapter/kegiatan/KegiatanAdapter.java
package com.gustu.logbook.main.adapter.kegiatan;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.RadioButton;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.gustu.logbook.R;
import com.gustu.logbook.main.model.kegiatan.Kegiatan;
import com.gustu.logbook.sharePreferences.SharedPrefUtil;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class KegiatanAdapter extends RecyclerView.Adapter<KegiatanAdapter.ViewHolder> {
    List<Kegiatan> kegiatanList;
    Context context;

    public void setContext(Context context) {
        this.context = context;
    }

    public KegiatanAdapter(List<Kegiatan> kegiatanList) {
        this.kegiatanList = kegiatanList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_kegiatan_dialog,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Kegiatan kegiatan = kegiatanList.get(position);
        holder.namaKegiatan.setText(kegiatan.getMKLNAMA());
        holder.linearLayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                SharedPrefUtil.saveString("nama_kegiatan",kegiatanList.get(position).getMKLNAMA());
                SharedPrefUtil.saveString("keterangan_kegiatan",kegiatanList.get(position).getMKLKETERANGAN());
            }
        });
        holder.radioButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                SharedPrefUtil.saveString("nama_kegiatan",kegiatanList.get(position).getMKLNAMA());
                SharedPrefUtil.saveString("keterangan_kegiatan",kegiatanList.get(position).getMKLKETERANGAN());
            }
        });
    }

    @Override
    public int getItemCount() {
        if (kegiatanList!=null){
            return kegiatanList.size();
        }
        return 0;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.radioKegiatan)
        RadioButton radioButton;
        @BindView(R.id.tvnamaKegiatan)
        TextView namaKegiatan;
        @BindView(R.id.layoutitemkegiatan)
        RelativeLayout linearLayout;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this,itemView);
        }
    }
}

File: app/src/main/java/com/gustu/logbook/main/adapter/kegiatan/KegiatanAdapter.kt
package com.gustu.logbook.main.adapter.kegiatan

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

import com.gustu.logbook.R
import com.gustu.logbook.main.model.kegiatan.Kegiatan
import com.gustu.logbook.sharePreferences.SharedPrefUtil
import kotlinx.android.synthetic.main.item_kegiatan_dialog.view.*

class KegiatanAdapter(internal var kegiatanList: List<Kegiatan>?) : RecyclerView.Adapter<KegiatanAdapter.ViewHolder>() {
    internal lateinit var context: Context
    internal var mCheckPosition = 0
    fun setContext(context: Context) {
        this.context = context
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_kegiatan_dialog, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val kegiatan = kegiatanList!![position]
        holder.namaKegiatan.text = kegiatan.mklnama
        holder.linearLayout.setOnClickListener {
            SharedPrefUtil.saveString("nama_kegiatan", kegiatanList!![position].mklnama!!)
            SharedPrefUtil.saveString("keterangan_kegiatan", kegiatanList!![position].mklketerangan)
            SharedPrefUtil.saveString("kode_kegiatan", kegiatanList!![position].mklkode!!)
        }

        holder.radioButton.isChecked = position == mCheckPosition
        holder.radioButton.setOnClickListener {
            if (position == mCheckPosition) {
                holder.radioButton.isChecked = false
                mCheckPosition = -1
            } else {
                mCheckPosition = position
                notifyDataSetChanged()
            }
            SharedPrefUtil.saveString("nama_kegiatan", kegiatanList!![position].mklnama!!)
          SharedPrefUtil.saveString("keterangan_kegiatan", kegiatanList!![position].mklketerangan)
            SharedPrefUtil.saveString("kode_kegiatan", kegiatanList!![position].mklkode!!)
        }
    }

    override fun getItemCount(): Int {
        return if (kegiatanList != null) {
            kegiatanList!!.size
        } else 0
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var radioButton = itemView.radioKegiatan
                internal var namaKegiatan=itemView.tvnamaKegiatan
        internal var linearLayout = itemView.layoutitemkegiatan
    }
}
